### DISCUSSION TITLE
Display already uploaded image using Upload field

### DISCUSSION THREAD

**Discussion by thewebartisan7** [orchidsoftware/platform#2028] [Q&A] [answered]
I am using upload field with group like:

```php
            Upload::make('invites')
                ->title('Invites')
                ->groups('invites')
```

And everything works fine when creating new records.

How to display already uploaded images when in edit mode, I mean to display like when you upload something like:

<img width="423" alt="Screenshot 2021-12-27 at 11 38 34" src="https://user-images.githubusercontent.com/57477934/147463874-0238683a-2fe9-45b5-ac8c-888071b6fa2d.png">

**Comment by thewebartisan7** [comment]
I found that there is a hidden modal for search already uploaded files, but no way to open it.

<img width="983" alt="Screenshot 2021-12-27 at 13 05 30" src="https://user-images.githubusercontent.com/57477934/147470240-21e0f05b-61e0-42fa-aea9-01df25169b78.png">

and probably the class enter-scale should be center-scale

**Comment by thewebartisan7** [comment, marked as answer]
I was able to solve this but would be good to have something in the docs about it.

```php
// In the model
    public function imagesInvite(): MorphToMany
    {
        return $this->attachment('exposition-invite');
    }

// In the screen query() method
$this->exposition->load('imagesInvite');


// In the layout
            Upload::make('exposition.imagesInvite')
                ->title('Invites')
                ->groups('exposition-invite')
```

The name of input upload should match the data from model, in my case exposition.imagesInvite

**Reply by tabuna** [reply]
Hi, you can find a description of this process on the tutorial page https://orchid.software/en/docs/quickstart-files
If you think it needs to be added elsewhere, please do so. Each documentation page has an edit button.

**Reply by thewebartisan7** [reply]
Thanks for your reply, I much appreciate.

Yes I follow that docs and the other one about attachment, and in the end that helped me sort out this. However I would add a bit more when working with groups of attachment, since you mention about "post.attachment" without group. Also I would add this relation in the model of Post example, as well an example of relation when working with group, example a post with image attachment group and with video attachment group. I can update docs if you agree by using edit button you mention.

I have also difficulty to sort out when saving more than one group of attachment on the same form. I am using tab layout and each tab has three different group of attachment.

So firstly I do like so:

```php
        $model->attachment('group-one')->sync(
            $request->input('exposition.group-one', [])
        );

        $model->attachment('group-two')->sync(
            $request->input('exposition.group-two', [])
        );
```

But second sync "group-two" delete also "group-one".  And I also try `syncWithoutDetaching`, but this didn't delete the removed attachment. So I used below solution:

```php
        $model->attachment()->sync(
            array_merge(
                $request->input('exposition.group-one', []),
                $request->input('exposition.group-two', [])
            )
        );
```
